### PR TITLE
Refactor intl utils

### PR DESCRIPTION
**Pull request by CarlosNZ**
Bit offtrack here, but just doing this quickly as discussed [here](https://github.com/openmsupply/open-msupply/pull/1563#discussion_r1171935055)

Just refactors the `IntlUtils` into a single hook `useIntlUtils` that returns all relevant values and methods, namely: 

**Values**
- defaultLanguage: 'en',
- isRtl,
- currentLanguage,
- languageOptions,
- currentLanguageName,

**Methods**
- changeLanguage,
- getLocaleCode,
- getUserLocale,
- setUserLocale,
- getLocalisedFullName,

A couple of the existing methods were no longer needed, or could be simplified too. (in particular, there's no need to return the whole `i18n` object now. I think it's tidier and more consistent now.

> the functions aren't all hooks in the IntlUtils tho - so I would just expose the hook based items. Otherwise you are forcing the consumer to use the utils as a hook, and that might be annoying? I guess check current usage and see.

Yes, all usages are components which were mostly already using one of the hooks, or easily can. In the unlikely event we need access to one of these values from a non-React element, we can just pass it in in from the caller. 

# 🧪 How has/should this change been tested? 

- Can still switch language
- Uses default language if none specified in local storage
- existing locale-based functionality unchanged

Hope I haven't broken anything 😳

### CODE REVIEW

File: client/packages/common/src/intl/utils/IntlUtils.ts [hunk 1]
```diff
@@ -62,28 +49,47 @@ export const IntlUtils = {
       throw new Error(`Language '${language}' not supported`);
     }
     return 'en';
-  },
-  languageOptions,
-  getLanguageName: (language: string) =>
-    languageOptions.find(option => option.value === language)?.label,
```
Comment (CarlosNZ) [inline client/packages/common/src/intl/utils/IntlUtils.ts:68]: Usage only ever returned the current language `getLanguageName(i18n.language)`, so makes it simpler to just have the hook return `currentLanguage` as a value.

File: client/packages/common/src/intl/utils/IntlUtils.ts [hunk 2]
```diff
@@ -62,28 +49,47 @@ export const IntlUtils = {
...
+
+  const getLocalisedFullName = (
+    firstName: StringOrEmpty,
+    lastName: StringOrEmpty
+  ) => getFullName(language, firstName, lastName);
+
+  return {
+    // TODO: When the server supports a query to find the deployments
+    // default language, use a query to fetch it.
+    defaultLanguage: 'en',
```
Comment (mark-prins) [inline client/packages/common/src/intl/utils/IntlUtils.ts:79]: I couldn't find a usage of this - wondering why it is added?
The default language would be more useful in the IntlContext, where the `fallbackLng` is set. Though I doubt that is helpful either.
Also not sure about having a TODO here for fetching a default from the API - that is better at the host / app level and not in a helper function like this.

**Review by mark-prins — APPROVED** [review]
👍 very nice thanks!
Just one quibble about the default language. and I've added a commit to fix the jest tests.

File: client/packages/common/src/intl/utils/IntlUtils.ts (hunk 2 above)
Comment (CarlosNZ) [reply client/packages/common/src/intl/utils/IntlUtils.ts]: Yeah, it's not actually being used at all, but it was there in the original, so I left it in:
```ts
  // TODO: When the server supports a query to find the deployments
  // default language, use a query to fetch it.
  useDefaultLanguage: (): SupportedLocales => {
    return 'en';
  },
```

I'll get rid of it, keep it simple. :) 19382297d

**Pull request merged**